### ISSUE TITLE
Continuous run in play projects

### ISSUE THREAD

**Issue by decapo01** [inkytonik/atom-sbt#28]
Great package!  It's nice to not have to fire up intellij.  Will there be support for continuous running (`~run`) of a project?

**Comment by inkytonik** [comment]
Thanks for the feedback! Actually, the package is supposed to deal with continuous commands. Is it not working for you?

**Comment by decapo01** [comment]
It's currently working for continuous compile but not continuous run.  I just updated to version 0.13 yesterday.

**Comment by inkytonik** [comment]
Ah, right. I'll take a look as soon as I can.

**Comment by inkytonik** [comment]
Can you please provide more detail on what goes wrong for you? Continuous runs seem to be working for me. If possible, can you show me the command you are running and what the output is? 

I checked using a command that did a ~runMain and it seemed to be working fine. I.e., when I made changes in the code, it rebuilt, re-ran and showed the updated output. If I introduced errors the compile generated them and the editor showed the errors as I would expect.

**Comment by decapo01** [comment]
the command that I am using is ~run.

I introduced errors and again nothing is picked up in atom.

Here is a screen shot

![screenshot from 2018-05-02 19-30-18](https://user-images.githubusercontent.com/6868296/39556750-dd3d9104-4e3f-11e8-8d09-593cf73f127c.png)

here is a screen shot when i run ~compile

![screenshot from 2018-05-02 19-31-56](https://user-images.githubusercontent.com/6868296/39556762-ed73eece-4e3f-11e8-8f17-563e242603d2.png)


you can see the red dot and the squiggly marks in the file explorer.  There is also an entry in the linter.

This is a playframework application also.  The sbt.version fo this project  is 1.1.2 and the version of the sbt atom package is 0.13.0.  Let me know if I can help out more.

**Comment by inkytonik** [comment]
Thanks for the info. The plugin is looking for the message that sbt outputs when it's finished processing a continuous command: "1. Waiting for source changes... (press enter to interrupt)" where the number gets incremented each time. The output from the command will only be processed when it sees that waiting message.

So, it seems that in the ~compile case the message appears but in the ~run case it doesn't. Unfortunately, I can't see the end of the output in your screenshots. Do you get the waiting message?

**Comment by decapo01** [comment]
in the run scenario the console does not get to "waiting for sources to change" part

![screenshot from 2018-05-02 22-12-29](https://user-images.githubusercontent.com/6868296/39559805-2e947afc-4e56-11e8-903f-6726fa536190.png)

However, it does when "~compile" is ran

![screenshot from 2018-05-02 22-13-33](https://user-images.githubusercontent.com/6868296/39559813-3fc99b04-4e56-11e8-8e87-88bafc0d2b1f.png)

Just to note this is the basic play application from the sbt g8 seed.

**Comment by inkytonik** [comment]
Ok, thanks. I've tried that project created with

    sbt new playframework/play-scala-seed.g8

Indeed the ~run command doesn't output the tag line that the plugin is looking for. The log below shows that ~compile is fine but ~run is not.

It's not immediately clear what to do to fix this. In some settings the "Compilation failed" message can be used as the tag line, but in theory that message can be turned off. That is why the plugin matches on prompts and the "Waiting for source changes" message which should be there.

It's not clear why the Waiting message is not printed in this case, since it seems to be ok for non-play applications that I've tried. Maybe there is something in the play project configuration or run task that could be varied to make the behaviour more compatible with the default sbt behaviour..

```
[play-scala-seed] $ ~compile
[success] Total time: 0 s, completed 03/05/2018 4:51:08 PM
1. Waiting for source changes... (press enter to interrupt)
[info] Compiling 1 Scala source to /Users/asloane/Downloads/play-scala-seed/target/scala-2.12/classes ...
[error] /Users/asloane/Downloads/play-scala-seed/app/controllers/HomeController.scala:12:35: ':' expected but identifier found.
[error] class HomeController @Inject()(cc ControllerComponents) extends AbstractController(cc) {
[error]                                   ^
[error] /Users/asloane/Downloads/play-scala-seed/app/controllers/HomeController.scala:24:2: identifier expected but eof found.

[error] }
[error]  ^
[error] two errors found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 0 s, completed 03/05/2018 4:51:13 PM
2. Waiting for source changes... (press enter to interrupt)
[success] Total time: 0 s, completed 03/05/2018 4:51:17 PM
3. Waiting for source changes... (press enter to interrupt)

[play-scala-seed] $ ~run

--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000

(Server started, use Enter to stop and go back to the console...)

[success] Compiled in 205ms
[info] Compiling 1 Scala source to /Users/asloane/Downloads/play-scala-seed/target/scala-2.12/classes ...
[error] /Users/asloane/Downloads/play-scala-seed/app/controllers/HomeController.scala:12:35: ':' expected but identifier found.
[error] class HomeController @Inject()(cc ControllerComponents) extends AbstractController(cc) {
[error]                                   ^
[error] /Users/asloane/Downloads/play-scala-seed/app/controllers/HomeController.scala:24:2: identifier expected but eof found.
[error] }
[error]  ^
[error] two errors found
[error] (Compile / compileIncremental) Compilation failed
```

**Comment by laughedelic** [comment]
I think this is because Play has its own file watcher:

> --- (Running the application, **auto-reloading is enabled**) ---

So it doesn't rely on sbt's `~`.

**Comment by inkytonik** [comment]
Yep, so that would be it. Thanks for letting us know. Then at this point it's not clear what can be done to fix this at the Atom end. There appears to be no guaranteed message that appears when the step has finished executing.
